### PR TITLE
modules/openstack: Create custom sshd config

### PR DESCRIPTION
**Pull request by lblackstone**
- Disable root login
- Restrict to public key auth

### CODE REVIEW

**Comment by lblackstone** [comment]
I was seeing a ton of failed ssh attempts by bots, so I locked down the sshd config as shown in [these docs](https://coreos.com/os/docs/latest/customizing-sshd.html). I verified the change for openstack, but it might make sense to make the same change for the other providers as well.

**Comment by Quentin-M** [comment]
ok to test

File: modules/openstack/nodes/ignition.tf [hunk 1]
```diff
@@ -125,3 +126,13 @@ data "ignition_systemd_unit" "tectonic" {
...
   content = "${var.tectonic_service}"
 }
+
+data "ignition_file" "sshd" {
+  filesystem = "root"
+  path       = "/etc/ssh/sshd_config"
+  mode       = 0600
+
+  content {
+    content = "UsePrivilegeSeparation sandbox\nSubsystem sftp internal-sftp\nPermitRootLogin no\nAllowUsers core\nPasswordAuthentication no\nChallengeResponseAuthentication no"
```
Comment (s-urbaniak) [inline modules/openstack/nodes/ignition.tf]: I suggest to use the heredoc format like
```
content = <<EOF
UsePrivilegeSeparation sandbox
...
EOF
```

**Comment by s-urbaniak** [comment]
I am unsure about enforcing those settings for all platforms though as there might be users with different setups/users for ssh. (/cc @alexsomesan) Maybe @euank has a bit more thoughts/reasoning about the recommended setting in the Tectonic use case.

**Comment by lblackstone** [comment]
@s-urbaniak Updated to use heredoc notation.

**Comment by lblackstone** [comment]
@euank Thanks for the feedback. I'll push the revised config shortly.

I noticed the failed login attempts in the sshd unit logs in journalctl while I was trying to debug an unrelated issue.

File: modules/openstack/nodes/ignition.tf [hunk 2]
```diff
@@ -125,3 +126,21 @@ data "ignition_systemd_unit" "tectonic" {
...
+  mode       = 0600
+
+  content {
+    content = <<EOF
+UsePrivilegeSeparation sandbox
+Subsystem sftp internal-sftp
+
+PermitRootLogin no
+AllowUsers core
+PasswordAuthentication no
```
Comment (lblackstone) [inline modules/openstack/nodes/ignition.tf]: @euank I pulled this example config from https://coreos.com/os/docs/latest/customizing-sshd.html

Might be worth updating those docs if you have revision suggestions there.

File: modules/openstack/nodes/ignition.tf [hunk 3]
```diff
@@ -125,3 +126,21 @@ data "ignition_systemd_unit" "tectonic" {
...
+
+data "ignition_file" "sshd" {
+  filesystem = "root"
+  path       = "/etc/ssh/sshd_config"
+  mode       = 0600
+
+  content {
+    content = <<EOF
+UsePrivilegeSeparation sandbox
+Subsystem sftp internal-sftp
```
Comment (lblackstone) [inline modules/openstack/nodes/ignition.tf]: Looks like this is already [the default](http://man.openbsd.org/sshd_config#UseDNS)

**Comment by lblackstone** [comment]
@euank Any other changes you'd like to see?

**Comment by lblackstone** [comment]
ping @euank

**Comment by lblackstone** [comment]
@s-urbaniak @euank Any other changes you'd like to see? I'd like to get this merged if possible. Thanks!

**Comment by euank** [comment]
ok to test

**Comment by lblackstone** [comment]
@coreos/team-installer OpenStack only. Good to merge.

**Comment by s-urbaniak** [comment]
@lblackstone lgtm, thanks!

**Pull request merged**